core/node/auth/auth_impl.go

```diff
@@ -670,21 +691,64 @@ func (ca *chainAuth) isEntitledToChannel(ctx context.Context, cfg *config.Config
 	return isEntitled.IsAllowed(), nil
 }
 
-func (ca *chainAuth) getLinkedWallets(ctx context.Context, wallet common.Address) ([]common.Address, error) {
+func (ca *chainAuth) getLinkedWalletsUncached(
+	ctx context.Context,
+	_ *config.Config,
+	args *ChainAuthArgs,
+) (CacheResult, error) {
+	log := dlog.FromCtx(ctx)
+
+	wallets, err := entitlement.GetLinkedWallets(ctx, args.principal, ca.walletLinkContract, nil, nil, nil)
+	if err != nil {
+		log.Error("Failed to get linked wallets", "err", err, "wallet", args.principal.Hex())
+		return nil, err
+	}
+
+	return &linkedWalletCacheValue{
+		wallets: wallets,
+	}, nil
+}
+
+func (ca *chainAuth) getLinkedWallets(
+	ctx context.Context,
+	cfg *config.Config,
+	args *ChainAuthArgs,
+) ([]common.Address, error) {
 	log := dlog.FromCtx(ctx)
 
 	if ca.walletLinkContract == nil {
 		log.Warn("Wallet link contract is not setup properly, returning root key only")
-		return []common.Address{wallet}, nil
+		return []common.Address{args.principal}, nil
 	}
 
-	wallets, err := entitlement.GetLinkedWallets(ctx, wallet, ca.walletLinkContract, nil, nil, nil)
+	userCacheKey := newArgsForLinkedWallets(args.principal)
+	// We want fresh linked wallets when evaluating space and channel joins, key solicitations,
+	// and user scrubs, all of which request the Read permission.
```

Comment:
just confirming, joining a space checks Permission.Read and not Permission.JoinSpace?

Where does Permission.JoinSpace come into play? Does it at all?
I just looked this up and in Permissions.sol it's `Permissions.JoinSpace` whereas in permissions.go its `PermissionJoin`

Reply:
The Read check is actually a bug I'm fixing in another PR, we have a few of those in our auth code still. I'll add you as a reviewer.

The difference in string constant values for Join and JoinSpace probably needs to be fixed.

So we actually ought to be doing something more subtle here. The contract checks the Join permission and if the user qualifies, it issues the token. The node itself then doesn't need to check if the user also qualifies for the join, it just needs to check that the user has a membership token, which was issued by the contract, who is the ultimate decider of who joins the space. That's the PR in the works. I proactively put in logic that would be compatible with this change.